ui/app2/src/main/java/org/activityinfo/ui/client/page/config/DbListPresenter.java

```diff
@@ -212,6 +216,70 @@ public void onSuccess(Void result) {
         });
     }
 
+    public void onTransfer() {
+        DatabaseTransferForm form = new DatabaseTransferForm(selection, dispatcher);
+        final FormDialogImpl dialog = new FormDialogImpl(form);
+        dialog.setWidth(400);
+        dialog.setHeight(170);
+        dialog.setHeadingText(I18N.CONSTANTS.transferDatabase());
+
+        form.getCancelButton().addClickHandler(cancelTransfer -> {
+            dispatcher.execute(new CancelDatabaseTransfer(selection.getId()), new SuccessCallback<VoidResult>() {
+                @Override
+                public void onSuccess(VoidResult result) {
+                    eventBus.fireEvent(AppEvents.SCHEMA_CHANGED);
+                    loader.load();
+                    dialog.hide();
+                }
+            });
+        });
+
+        dialog.show(new FormDialogCallback() {
+            @Override
+            public void onValidated() {
+                form.getUser(new AsyncCallback<UserProfileDTO>() {
```

Comment:
Why do all this work on the client? Just send the email to the server along with the transfer database request and let the server look up the email address.

Reply:
Will change in next iteration.